dart/lib/src/sentry_envelope_item.dart

```diff
@@ -109,16 +133,11 @@ class SentryEnvelopeItem {
 class _CachedItem {
   _CachedItem(this._dataFactory);
 
-  final FutureOr<List<int>> Function() _dataFactory;
+  final Future<List<int>> Function() _dataFactory;
   List<int>? _data;
 
   Future<List<int>> getData() async {
```

Comment:
@denrase I'd like a review especially here.
From my understanding we were not awaiting for the _dataFactory call, but only for its result. 
This means the factory could be called multiple times: I added a print in the CachedItems functions of the envelope items and they were actually called several times.
So i decided to "force" the factory to be a `Future` instead of `FutureOr`, and the attachment factory (the only one that was `FutureOr`) to be `async`.
Is my understanding correct?

Reply:
the intent is clear
I guess it was a problem because we were doing `_dataFactory()` which started the function and after that we waited for its response if it was a future. But we were starting it everytime from what i could tell